azure_auth/handlers.py

```diff
@@ -97,7 +98,26 @@ def authenticate(self, token: dict) -> UserModel:
             user.last_name = attr if (attr := azure_user["surname"]) else ""
             user.save()
 
-        # TODO: Handle groups
+        # Syncing azure token claim roles with django user groups
+        # A role mapping in the AZURE_AUTH settings is expected.
+        role_mappings = settings.AZURE_AUTH.get("ROLES")
+        if role_mappings:
+            # Check if id_token_claims and roles are available
+            if token.get("id_token_claims", None):
+                azure_token_roles = token.get("id_token_claims", None).get("roles", None)
+
+                for role, group_name in role_mappings.items():
+                    # all groups are created by default if they not exist
+                    django_group = Group.objects.get_or_create(name=group_name)[0]
+                    
+                    if role in azure_token_roles:
```

Comment:
I think this could potentially error with a `NoneType is not iterable` if the token claims don't include `roles` as `azure_token_roles` would be `None` based on line 107!

Reply:
Thanks for your review and feedback. Really appreciate!
I tried to already keep the conditional checks as lean as possible.

I will have a look onto this and provide fix.

Reply:
> I think this could potentially error with a `NoneType is not iterable` if the token claims don't include `roles` as `azure_token_roles` would be `None` based on line 107!

Since `azure_token_roles` is moved up and checked before this get's executed this should not happen.

Reply:
Yeah sorry I made this comment before the other main one!

Reply:
I provided a new version. Kindly have a look into it.